src/app/components/SettingsSaveChanges.tsx

```diff
@@ -0,0 +1,69 @@
+import { Center, Button } from "@chakra-ui/react";
+import { gasHawkClient } from "../interface/gashawk-backend";
+import { userDataByAddress } from "../store/atoms/userDataByAddress";
+import { useRecoilState } from "recoil";
+import { defaultRpcType } from "./SettingsFlashbots";
+
+export type toastStatusType =
+  | "info"
+  | "warning"
+  | "success"
+  | "error"
+  | "loading"
+  | undefined;
+
+type SettingsSaveChangesProps = {
+  autoQueue: boolean;
+  sliderValue: number;
+  holdBackIfBaseFeeExceedsBfCreatedAt: boolean;
+  preventStuckInMempool: boolean;
+  probeTxBeforeSubmission: boolean;
+  flashbots: defaultRpcType;
+  showToast: (message: string, status: toastStatusType) => void;
+};
+
+export const SettingsSaveChanges: React.FC<SettingsSaveChangesProps> = ({
+  autoQueue,
+  sliderValue,
+  holdBackIfBaseFeeExceedsBfCreatedAt,
+  preventStuckInMempool,
+  probeTxBeforeSubmission,
+  flashbots,
+  showToast,
+}) => {
+  const [{ address: user }] = useRecoilState(userDataByAddress);
+
+  const handleSaveChanges = async () => {
+    if (user) {
+      try {
+        const res = await gasHawkClient.userSettingsUpdate(user, {
+          autoQueue: autoQueue,
+          defaultDeadlineDuration: sliderValue,
+          holdBackIfBaseFeeExceedsBfCreatedAt:
+            holdBackIfBaseFeeExceedsBfCreatedAt,
+          preventStuckInMempool: preventStuckInMempool,
+          probeTxBeforeSubmission: probeTxBeforeSubmission,
+          defaultRpc: flashbots,
+        });
```

Comment:
```suggestion
        const res = await gasHawkClient.userSettingsUpdate(user, {
          autoQueue,
          defaultDeadlineDuration: sliderValue,
          holdBackIfBaseFeeExceedsBfCreatedAt,
          preventStuckInMempool,
          probeTxBeforeSubmission,
          defaultRpc: flashbots,
        });
```

and if you rename `flashbots` to `defaultRpc` and `sliderValue` to `defaultDeadlineDuration`, it will be even shorter.

Reply:
Renamed variables. [84de6b4](https://github.com/sohjag/gashawk-app-ext/pull/5/commits/84de6b445ae1b6ed9b93ce6a0ec70e5f2bfd2da2)